### DISCUSSION TITLE
Initial Questions in relation to using Elsa in an ETL context

### DISCUSSION THREAD

**Discussion by GoOnYahGoodThing** [elsa-workflows/elsa-core#2867] [General]
Hello,
I'm looking at possible solutions to help us achieve some pretty straightforward ETL processing goals.
Basically the monitoring and controlling of ETL jobs.

On the face of it Elsa has some really impressive facilities which I am hoping would allow us to achieve the basic goals and give us loads of scope for expanding the workflows and doing some more sophisticated processing.

Before I dive into installing and trialling I was hoping someone could maybe help me understand if our goals are in line with the uses that Elsa is intended for.

Our most common ETL tasks involve extracting data (from files or oracle database tables), trasforming it to JSON and sending it to a REST api endpoint.

Our initial goal is to have a dashboard to **monitor** the progess of these tasks and allow us to **control** them (start, stop, schedule)

Some points to note:
- The files can contain up to 4 million rows, so the task could take a couple of days to run.
- If resources come under pressure we might want to pause the task, or stop it completely.
- At other times we might want to schedule the task to start at, say, 10pm on a Friday night and stop again at 10pm on Sunday night.
- The tasks need to have different command line (or other) params depending on the run.

Do Elsa workflows map to these kinds of tasks? 
Does the Dashboard allow users to stop or start tasks manually?
Does the Dashboard have a GUI for setting schedules ?

Thanks for your time.

**Comment by sfmskywalker** [comment]
Hi @GoOnYahGoodThing,

Elsa works well to orchestrate a process such as starting/scheduling a job and then continuing with other logic once the job completes.

Although you could run these long-running tasks from a custom activity if you want, it might be better to run these jobs external from the workflow itself using a proper job manager such as Hangfire. When the job fails, Hangfire will automatically retry the job several times

Alternatively, you could decide to process individual records within the workflow itself in a looping construct. This has the added benefit of being able to maintain a current index/cursor of the record being processed in case the process crashes for some reason - then it wouldn't have to start over.

The dashboard however doesn't offer much control at the moment when it comes to stopping/starting tasks manually.

But I can imagine a new module called e.g. "Jobs" that extends the dashboard with a job management UI allowing the user to start/stop jobs at will. This would happen independently from any workflows - the workflows simply wait for jobs to complete before they continue, for example. 

However, this doesn't help you now. I know Hangfire has a dashboard that let's you monitor jobs, retry failed ones and has APIs to start jobs (I don't think the dashboard exposes this however) which might be interesting to look at. As a bonus, it can be used together with Elsa, where Elsa becomes an orchestrator of Hangfire Jobs.

**Reply by GoOnYahGoodThing** [reply]
Right. Thanks very much for taking the time to answer my questions.
I better go look at Hangfire then!
Thanks!
